### ISSUE TITLE
Looks like the sort filter breaks when the sorted field contains a null.

### ISSUE THREAD

**Issue by mynoc96** [GhostManager/Ghostwriter#524]
**Describe the bug**
Jinja sort filter fails during report generation if the field being sorted for one of the sorted elements in the list contains a null.

I haven't validated in code, but I don't have any less-than comparisons (<) in my report, but that's the error, and when I remove the sort filter, the template works properly. I'm hoping it isn't an issue in an upstream project. I think a valid work-around is to create a new list using variable assignment ("set"), and populate it with values from the original list, properly replacing the null with "", then sorting and using that new list.

**To Reproduce**
1. Create a project and create an oplog entry with the tool field set to "something"
2. Create another oplog, but don't enter any data.
3. Create a for loop in a report like the following:

```
{% for log in logs %}
{% for entry in log.entries|sort(attribute=”tool”) %}
{# Do something meaningful #}
{% endfor %}
{% endfor %}
```

5. Add the report template.
6. It should not generate an error here.
7. Create a report for the project.
8. Observe that it errors out with an error about a "<" comparison. 

**Expected Behavior**
The sort should ignore nulls or treat them like a "".

**Server Specs:**
 - OS: Ubuntu 22.04 on WSL.
 - Docker 24.0.7
 - Ghostwriter 4.2.5

**Comment by chrismaddalena** [comment]
Hey @mynoc96, Jinja2's `sort` filter will encounter an error if it runs into a null value. The error you see some from Jinja2 trying to compare the values. Jinja2 use's Python's `sorted()` function:

https://docs.python.org/3/library/functions.html#sorted

Jinja2 doesn't have a built-in way to handle/catch the possible empty value. If this were a direct call to `sorted()`, you could use `key` arg and `lambda` to handle the empty values.

It's possible to check for an empty value and set it to a blank string or something else, but that requires a line like `entry.__setitem__("tool", "")`, but access to internal methods is disallowed to prevent malicious Python execution inside templates.

One option is ensuring your log entries all have a `tool` value if you want to sort on that. Another option is sorting based on a different value you know won't be blank.

We can look into doing something like creating a custom filter that ensures any null values are set to blank strings for the sake of a built-in filter like `sort`.

**Comment by mynoc96** [comment]
Thanks for the thorough response. I realize that there is a business logical difference between the value being null versus "" (i.e., the user didn't enter a value), and that it's easier to check existence with {% if field %} versus {% if field != "" %}, but some fields in the data structures initialize to "", whereas some initialize to null. Would it be possible to initialize all nulls as "" for report generation?

**Comment by chrismaddalena** [comment]
We will look into that. For now, I've added a `replace_blanks` filter to help. It may be useful for custom JSON field values even if we change empty fields to always be `""`.

**Comment by mynoc96** [comment]
You're a rock star. Love the solution. Thanks!

**Comment by mynoc96** [comment]
I like that idea. It may cause issues with current users upgrading if their template logic is:

```
Some Header
{%p if not some_field %}
N/A
{%p else %}
{{ some_field }}
{%p endif %}
```

But, that's a "your call" issue. :-)

**Comment by chrismaddalena** [comment]
I am considering that, but it should be alright. The `{% if not X %}` statement should return `true` if `X` is `""` or `null`. I can't think of any possible statements that will behave differently if a `null` becomes `""`.

**Comment by chrismaddalena** [comment]
This is resolved in v4.3.1 from yesterday.
